### PR TITLE
Add per game configuration options

### PR DESCRIPTION
**Pull request by GPUCode**
This is the first in a series of yuzu backport PRs that aim to modernize our frontend. This includes the ability to store settings per game. It also incurs some other changes, for example the alternative emulation speed slider was removed since I found it redundant when it can be set per game (can be brought back if desired though). Config files are stored, similarly to yuzu in config/custom/<title_id>.ini.

- [x] Move most options to per-game menu
- [ ] ~~Move cheats to per-game menu~~ (will be done in another PR)
- [x] Cleanup code and fix all build targets

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6187)

### CODE REVIEW

File: src/citra_qt/configuration/configure_per_game.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    ui->tabWidget->addTab(system_tab.get(), tr("System"));
+    ui->tabWidget->addTab(graphics_tab.get(), tr("Graphics"));
+    ui->tabWidget->addTab(audio_tab.get(), tr("Audio"));
+
+    setFocusPolicy(Qt::ClickFocus);
+    setWindowTitle(tr("Properties"));
+    // remove Help question mark button from the title bar
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+    // addons_tab->SetTitleId(title_id);
```
Comment (FearlessTobi) [inline src/citra_qt/configuration/configure_per_game.cpp:45]: Should this be removed?

File: src/citra_qt/configuration/configure_per_game.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                &ConfigurePerGame::HandleApplyButtonClicked);
+    }
+
+    LoadConfiguration();
+}
+
+ConfigurePerGame::~ConfigurePerGame() = default;
+
+void ConfigurePerGame::ApplyConfiguration() {
+    // addons_tab->ApplyConfiguration();
```
Comment (FearlessTobi) [inline src/citra_qt/configuration/configure_per_game.cpp:62]: dito for all other occurrences

File: src/citra_qt/main.cpp [hunk 3]
```diff
@@ -1477,6 +1484,19 @@ void GMainWindow::OnGameListShowList(bool show) {
     game_list_placeholder->setVisible(!show);
 };
 
+void GMainWindow::OnGameListOpenPerGameProperties(const QString& file) {
+    u64 title_id{};
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp:1488]: this can be moved right before the if-statement

File: src/common/settings.cpp [hunk 4]
```diff
@@ -13,24 +14,35 @@
 #include "core/hle/service/ir/ir_rst.h"
 #include "core/hle/service/ir/ir_user.h"
 #include "core/hle/service/mic_u.h"
-#include "core/settings.h"
 #include "video_core/renderer_base.h"
 #include "video_core/video_core.h"
 
 namespace Settings {
 
+[[nodiscard]] std::string_view ToString(AudioEmulation emulation) {
```
Comment (FearlessTobi) [inline src/common/settings.cpp:22]: This should have a more descriptive name or maybe be moved entirely into the Log_Settings function as a lambda.

File: src/common/settings.h [hunk 5]
```diff
@@ -0,0 +1,532 @@
...
+    SideBySide = 1,
+    Anaglyph = 2,
+    Interlaced = 3,
+    ReverseInterlaced = 4,
+    CardboardVR = 5
+};
+
+enum class AudioEmulation : u32 { HLE = 0, LLE = 1, LLEMultithreaded = 2 };
+
+namespace NativeButton {
```
Comment (FearlessTobi) [inline src/common/settings.h:55]: nit: should have a newline after namespace declaration

File: src/common/settings.h [hunk 6]
```diff
@@ -0,0 +1,532 @@
...
+    "button_select",
+    "button_debug",
+    "button_gpio14",
+    "button_zl",
+    "button_zr",
+    "button_home",
+}};
+} // namespace NativeButton
+
+namespace NativeAnalog {
```
Comment (FearlessTobi) [inline src/common/settings.h:113]: dito

File: src/citra_qt/configuration/configure_per_game.cpp [hunk 7]
```diff
@@ -0,0 +1,133 @@
+// SPDX-FileCopyrightText: Copyright 2020 yuzu Emulator Project
```
Comment (FearlessTobi) [inline src/citra_qt/configuration/configure_per_game.cpp:1]: In contrast to yuzu, we do not use SPDX copyright headers.

**Review by FearlessTobi — COMMENTED** [review]
Code looks very good already, just some small issues.

File: src/citra_qt/configuration/configure_per_game.cpp (hunk 1 above)
Comment (GPUCode) [reply src/citra_qt/configuration/configure_per_game.cpp:45]: I intended to add cheats in the per game menu, but they require the game to be running so I'm not sure if I go ahead with that

File: src/citra_qt/configuration/configure_per_game.cpp (hunk 7 above)
Comment (GPUCode) [reply src/citra_qt/configuration/configure_per_game.cpp:1]: The code has been taken mostly from yuzu so I think it's right to leave the copyright unmodified to credit the original source

Comment (FearlessTobi) [reply src/citra_qt/configuration/configure_per_game.cpp:1]: Yes, but please credit them in our usual style of copyright headers:
```
// Copyright 2020 yuzu Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.
```

Comment (GPUCode) [reply src/citra_qt/configuration/configure_per_game.cpp:1]: Done

**Review by FearlessTobi — APPROVED** [review]

**Comment by FearlessTobi** [comment]
Please fix clang-format.

**Comment by luc-git** [comment]
Hello here is a the fixed code format
 https://github.com/luc-git/citra/commit/9be3063cf1119dc4ea1323df98b2d36c2c0ab268

**Comment by luc-git** [comment]
If you use visual studio community IDE I recommend you to use Ctrl+K, Ctrl+D to format the file.

https://learn.microsoft.com/en-us/visualstudio/ide/reference/options-text-editor-c-cpp-formatting?view=vs-2022

**Pull request merged**